embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/caching/PayloadCachingServiceImpl.kt

```diff
@@ -0,0 +1,28 @@
+package io.embrace.android.embracesdk.internal.delivery.caching
+
+import io.embrace.android.embracesdk.internal.payload.Envelope
+import io.embrace.android.embracesdk.internal.payload.SessionPayload
+import io.embrace.android.embracesdk.internal.session.caching.PeriodicBackgroundActivityCacher
+import io.embrace.android.embracesdk.internal.session.caching.PeriodicSessionCacher
+
+class PayloadCachingServiceImpl(
+    private val periodicSessionCacher: PeriodicSessionCacher,
+    private val periodicBackgroundActivityCacher: PeriodicBackgroundActivityCacher
+) : PayloadCachingService {
+
+    override fun shutdown() {
+    }
+
+    override fun startCaching(isInBackground: Boolean, supplier: () -> Envelope<SessionPayload>?) {
```

Comment:
It seems pretty imprecise for this to be a parameter passed in, and there being a chance for the payload to not match what this value is.

I understand if this is what the current implementation does, but it probably makes sense for this component to derive the background-ness of the given session and do the caching base on that?